src/core/server/saved_objects/service/lib/collect_multi_namespace_references.ts

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+// @ts-expect-error no ts
+import { esKuery } from '../../es_query';
+
+import { LegacyUrlAlias, LEGACY_URL_ALIAS_TYPE } from '../../object_types';
+import type { ISavedObjectTypeRegistry } from '../../saved_objects_type_registry';
+import type { SavedObjectsSerializer } from '../../serialization';
+import type { SavedObject, SavedObjectsBaseOptions } from '../../types';
+import { getRootFields } from './included_fields';
+import { getSavedObjectFromSource, rawDocExistsInNamespace } from './internal_utils';
+import type {
+  ISavedObjectsPointInTimeFinder,
+  SavedObjectsCreatePointInTimeFinderOptions,
+} from './point_in_time_finder';
+import type { RepositoryEsClient } from './repository_es_client';
+
+/**
+ * When we collect an object's outbound references, we will only go a maximium of this many levels deep before we throw an error.
+ */
+const MAX_REFERENCE_GRAPH_DEPTH = 20;
+
+/**
+ * How many aliases to search for per page. This is smaller than the PointInTimeFinder's default of 1000.
```

Comment:
I chose 100 somewhat arbitrarily, for better performance based on advice in #93770.

That issue specifically mentions expensive queries > 1000 objects. Thinking through the use case for this function, it's not very important in terms of Kibana functionality overall (it will only be called when a user opens the Share to Space flyout), so I figured a page size that is another order of magnitude smaller would be appropriate here. 

If you agree that makes sense, I'd be happy to update the comment accordingly. Or if you think thats unnecessary, I can remove it 🤷‍♂️

Reply:
I'm fine with your choice of 100 -- updating the comment accordingly sounds good to me